core/primitives/src/test_utils.rs

```diff
@@ -427,6 +413,238 @@ impl BlockHeader {
             }
         }
     }
+
+    pub fn init(&mut self) {
+        match self {
+            BlockHeader::BlockHeaderV1(_) => unreachable!("old header should not appear in tests"),
+            BlockHeader::BlockHeaderV2(_) => unreachable!("old header should not appear in tests"),
+            BlockHeader::BlockHeaderV3(_) => unreachable!("old header should not appear in tests"),
+            BlockHeader::BlockHeaderV4(header) => Arc::make_mut(header).init(),
+            BlockHeader::BlockHeaderV5(header) => Arc::make_mut(header).init(),
+        }
+    }
```

Comment:
It's a good idea to have all variants covered, so that when the next one is added it won't go into the default case. That being said you can have all of them in a single pattern with `|`.

Reply:
yeah my intention was to have all variants explicitly listed. but I will use `|` to combine them into single mapping.